dumpgenerator.py

```diff
@@ -1090,7 +1090,7 @@ def makeXmlFromPage(page):
                 ))
             else:
                 revision.append(E.contributor(deleted="deleted"))
-            if 'comment' in rev:
+            if 'comment' in rev and rev["comment"]:
```

Comment:
Is this what special:export does?

Reply:
Yes, `special:export` and `dumpBackup.php` (*MW CLI*) both do it this way.  
You can verify this on [these test sites](https://github.com/mediawiki-client-tools/mediawiki-scraper/issues/7#issuecomment-1374490611).